enterprise/management/commands/create_enterprise_customer_admins.py

```diff
@@ -0,0 +1,85 @@
+"""
+Management command to create EnterpriseCustomerAdmin records for users with the enterprise_admin role.
+"""
+
+import logging
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from enterprise.models import (
+    EnterpriseCustomerAdmin,
+    EnterpriseCustomerUser,
+    SystemWideEnterpriseUserRoleAssignment,
+)
+from enterprise.utils import batch
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Create EnterpriseCustomerAdmin records for users with the enterprise_admin role.
+    """
+    help = 'Creates EnterpriseCustomerAdmin records for users with the enterprise_admin role'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Show what would be done without making changes',
+        )
+        parser.add_argument(
+            '--batch-size',
+            action='store',
+            dest='batch_size',
+            default=500,
+            help='Number of role assignments to process in each batch.',
+            type=int,
+        )
+
+    def handle(self, *args, **options):
+        dry_run = options['dry_run']
+        batch_size = options['batch_size']
+
+        if dry_run:
+            logger.info('DRY RUN - No changes will be made')
+
+        admin_role_assignments = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+            role__name='enterprise_admin'
+        ).select_related('user', 'enterprise_customer')
+
+        for role_assignments_batch in batch(admin_role_assignments, batch_size=batch_size):
```

Comment:
Awesome that you used `batch` here! That completely avoid problems with loading too many records into memory.